### PR TITLE
boards: arduino_nicla_vision: add Arduino Nicla Vision

### PR DESCRIPTION
**Pull request by uLipe**
Initial board support and documentation. 

Unfortunately the camera support would take some time, because there is no GC2145 driver yet on Zephyr, on the other hand it ispossible to find a OV26 or 5640 based camera module that matches with current connection since the DCMI for STM32 already landed in the upstream.

### CODE REVIEW

**Comment by uLipe** [comment]
Any hints regarding the failling test of the kernel/timer?

**Comment by uLipe** [comment]
It seems this PR is unrelated to the CI failure, the errors might be related to this:

https://github.com/zephyrproject-rtos/zephyr/issues/77496

**Review by fabiobaltieri — DISMISSED** [review]

**Comment by uLipe** [comment]
Just added the USB support, and tested with CDC, Mass and Console samples.

**Review by fabiobaltieri — DISMISSED** [review]

**Comment by uLipe** [comment]
Wifi also will take some time, I was trying to add it, but  there is a blocker on ST side:

https://github.com/zephyrproject-rtos/zephyr/discussions/77079

Once we get the SDIO proper support on STM32 , it will be possible to use the AIROC module of the Nicla Vision.

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: boards/arduino/nicla_vision/doc/img/arduino_nicla_vision.jpg
[the remark this answers is not in the document]
Comment (uLipe) [reply boards/arduino/nicla_vision/doc/img/arduino_nicla_vision.jpg:1]: I just dropped the file into the url above and replaced, it showed it was compressed. Is there any step I'm missing?

[the remark this answers is not in the document]
Comment (nordicjm) [reply boards/arduino/nicla_vision/doc/img/arduino_nicla_vision.jpg:1]: Compressed but it should be in webp, @kartben can give advice

[the remark this answers is not in the document]
Comment (uLipe) [reply boards/arduino/nicla_vision/doc/img/arduino_nicla_vision.jpg:1]: Changed to webp. Thanks.

**Review by nordicjm — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**